### ISSUE TITLE
Acquire accessToken for app for AzureB2C

### ISSUE THREAD

**Issue by michiproep** [AzureAD/microsoft-identity-web#2231]
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

2.10.0

### Web app

Sign-in users and call web APIs

### Web API

Protected web APIs call downstream web APIs

### Token cache serialization

Distributed caches

### Description

With version 2.10.0, one can still not get access tokens for confidential client application for azure b2c with
`
tokenAquisition.GetAccessTokenForAppAsync(ApiAppScope )
`
This also makes other apis like IDownstreamApi impossible to use.

It will throw
MsalClientException: WithTenantId can only be used when an AAD authority is specified at the application level.
/

### Reproduction steps

Hook your webApp to b2c and call your api with tokenAquisition.GetAccessTokenForAppAsync(ApiAppScope ).

=> Exception: MsalClientException: WithTenantId can only be used when an AAD authority is specified at the application level.

### Error message

MsalClientException: WithTenantId can only be used when an AAD authority is specified at the application level.

### Relevant code snippets

```csharp
There was a change with the lastest release (2.10.0) which I initially thought this would fix the problem but now there is second change which is quite strange and reintroduces the problem:

In TokenAcquisition.cs

// MSAL.net only allows .WithTenantId for AAD authorities. This makes sense as there should
            // not be cross tenant operations with such an authority.
            if (!mergedOptions.Instance.Contains(".ciamlogin.com"
#if NETCOREAPP3_1_OR_GREATER
                , StringComparison.OrdinalIgnoreCase
#endif
                ))
            { 
                   builder.WithTenantId(tenant);
            }
```

### Regression

hasn't worked in a long time, if ever

### Expected behavior

Whatever this means? 
" if (!mergedOptions.Instance.**Contains(".ciamlogin.com**" "

It should be something like
"if( IsAzureAD)"
since WithTenantId can only be used with AAD. All other cases may not use WithTenantId

**Comment by michiproep** [comment]
#2106 is related. not sure why this got closed

**Comment by michiproep** [comment]
@jennyf19 Any news on this?
The issue basically is "one cannot use AcquireTokenForApp with B2C since ConfidentialClient-setup always adds WtihTenantId"

**Comment by michiproep** [comment]
@jennyf19 Any news?

**Comment by jmprieur** [comment]
It should be fixed in 2.19.2 (there was a bug in MSAL.NET)
Can you please confirm that you are unblocked?

**Comment by michiproep** [comment]
Hi @jmprieur, you're right!
Finally!!! With the lastest version it works!!!
